libs/sdk-backend-tiger/src/backend/index.ts

```diff
@@ -193,8 +308,312 @@ export class TigerBackend implements IAnalyticalBackend {
+                        try {
+                            const result = await sdk.entities.createEntityDataSources({
+                                jsonApiDataSourceInDocument: sampleDataSource
+                                    ? { ...sampleDataSource }
+                                    : {
+                                          data: {
+                                              attributes: {
+                                                  name: "demo-ds",
+                                                  url: "jdbc:postgresql://localhost:5432/demo",
+                                                  schema: "demo",
+                                                  type: JsonApiDataSourceInAttributesTypeEnum.POSTGRESQL,
+                                                  username: "demouser",
+                                                  password: "demopass",
+                                              },
+                                              id: "demo-ds",
+                                              type: JsonApiDataSourceInTypeEnum.DATA_SOURCE,
+                                          },
+                                      },
+                            });
+                            return result.data.data.id;
+                        } catch (error) {
+                            throw convertApiError(error);
+                        }
+                    });
+                },
+                setPdmLayout: async (requestParameters: LayoutApiSetPdmLayoutRequest) => {
+                    return await this.authApiCall(async (sdk) => {
+                        try {
+                            await sdk.declarativeLayout.setPdmLayout(requestParameters);
+                        } catch (error) {
+                            throw convertApiError(error);
+                        }
+                    });
+                },
+                createWorkspace: async (id: string, name: string) => {
+                    return await this.authApiCall(async (sdk) => {
+                        try {
+                            const result = await sdk.entities.createEntityWorkspaces({
+                                jsonApiWorkspaceInDocument: {
+                                    data: {
+                                        attributes: {
+                                            name,
+                                        },
+                                        id,
+                                        type: JsonApiWorkspaceInTypeEnum.WORKSPACE,
+                                    },
+                                },
+                            });
+                            return result.data.data.id;
+                        } catch (error) {
+                            throw convertApiError(error);
+                        }
+                    });
+                },
+                createWorkspaceFromDocument: async (workspace: JsonApiWorkspaceInDocument) => {
```

Comment:
Can we maybe join `createWorkspace` and `createWorkspaceFromDocument` they seem relatively similar.

Reply:
I did not liked this one either. I dig more from where the document is provided (it's `gdc-public-demos` repo from where it is fetched via API call 🥲). I reused the `createWorkspace` function and deleted this one.

Reply:
Thanks :)